src/tfr_executive/src/autonomous_action_server.cpp

```diff
@@ -0,0 +1,128 @@
+         * The main autonomous procedure is long running and needs to be responsive to 
+         * preemption. This means that after making time consuming calls, for example 
+         * starting navigation, it needs to do so in a non blocking manner, and query
+         * for preemption from the user at a set interval. At this interval it is also 
+         * appropriate to ask for feedback from the time consuming processes, or check 
+         * for success. The definition of long running means any call that starts a 
+         * subsystem. 
+         * 
+         * Also this is an action server so it is responsible for reporting important 
+         * state changes and updates to the standard communication channel.
+         * 
+         * Judiciously following these guidelines, the skeleton of the main procedure is 
+         * as follows:
+         * 1. Run the localization subsystem.
+         * 2. Store the odometry information gathered.
+         * 3. Run the navigation subsystem with the drive to mining zone setting.
+         * 4. Run the digging action subsystem with the digging time, from the `get_digging_time` service.
+         * 5. Run the navigation subsystem with the return from mining zone option.
+         * 6. Run the dumping subsystem.
+         * 7. Put the system in teleop mode and await instructions.
+         * 
+         * ACTION COMPONENTS
+         * - Goal: none
+         * - Feedback: none
+         * - Result: none
+         */ 
+        void autonomousMission(const tfr_msgs::EmptyGoalConstPtr &goal)
+        {
+            ROS_INFO("Autonomous Action Server: mission started");
+
+            ROS_INFO("Autonomous Action Server: commencing localization");
+            ROS_INFO("Autonomous Action Server: localization finished");
+
+            ROS_INFO("Autonomous Action Server: commencing navigation");
+            ROS_INFO("Autonomous Action Server: navigation finished");
+
+            ROS_INFO("Autonomous Action Server: retrieving digging time");
+            tfr_msgs::DurationSrv digging_time;
+            ros::service::call("digging_time", digging_time);
+            ROS_INFO("Autonomous Action Server: digging time retreived %f",
+                    digging_time.response.duration.toSec());
+
+            ROS_INFO("Autonomous Action Server: commencing digging");
+            ROS_INFO("Autonomous Action Server: digging finished");
+
+            ROS_INFO("Autonomous Action Server: commencing navigation");
+            ROS_INFO("Autonomous Action Server: navigation finished");
+
+            ROS_INFO("Autonomous Action Server: commencing dumping");
+            ROS_INFO("Autonomous Action Server: dumping finished");
+
+            tfr_msgs::EmptyResult result{};
+            server.setSucceeded(result);
+            ROS_INFO("Autonomous Action Server: mission finished");
+        }
+
+        actionlib::SimpleActionServer<tfr_msgs::EmptyAction> server;
+        ros::Duration frequency;
```

Comment:
What is `frequency` going to be used for?

Reply:
How fast we check for preemption.

Reply:
I'll add in a comment for clarity

Reply:
I also noticed I only did this the autonomous server, I propagated the change out to the teleop as well.